src/engine/test/source/builder/helperParser_test.cpp

```diff
@@ -0,0 +1,53 @@
+#include <gtest/gtest.h>
+
+#include <re2/re2.h>
+
+#include "helperParser.hpp"
+
+using HelperParserT = std::tuple<bool, std::string, builder::internals::HelperToken>;
+class HelperParserTest : public ::testing::TestWithParam<HelperParserT>
+{
+};
+
+TEST_P(HelperParserTest, parse)
+{
+    auto &[shouldPass, input, expected] = GetParam();
+
+    auto result = builder::internals::parseHelper(input);
+
+    if (shouldPass)
+    {
+        ASSERT_TRUE(std::holds_alternative<builder::internals::HelperToken>(result));
+        ASSERT_EQ(std::get<builder::internals::HelperToken>(result).name, expected.name);
+        ASSERT_EQ(std::get<builder::internals::HelperToken>(result).args, expected.args);
+    }
+    else
+    {
+        ASSERT_TRUE(std::holds_alternative<base::Error>(result));
+    }
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    Builder,
+    HelperParserTest,
```

Comment:
I would add some cases in order to clear out the parenthesis behavior:
```
HelperParserT(true, "test(arg1, ( arg2)", {.name = "test", .args = {"arg1", "( arg2"}}),  // Testing escaped parenthesis
HelperParserT(true, "test(arg1, ) arg2)", {.name = "test", .args = {"arg1", ") arg2"}}),  // Testing escaped parenthesis
HelperParserT(true, "test(arg1, ) arg2))))", {.name = "test", .args = {"arg1", ") arg2)))"}}),  // Testing escaped parenthesis
```

Reply:
I agree